src/css/styles.css

```diff
@@ -9,6 +9,7 @@ html, #web-monitoring-ui-root, #web-monitoring-ui-root > div {
   display: flex;
   height: 100%;
   flex-direction: column;
+  overflow: scroll;
```

Comment:
I don’t think you should need to declare an overflowing region here at all, should you? If you do, this should probably be `auto` and not `scroll`, otherwise you get permanent, unnecessary scrollbars:

![scrollbars](https://user-images.githubusercontent.com/74178/33252132-19f442c0-d2f1-11e7-8d58-f2a6afe0963d.png)

Reply:
Yea, we need `overflow` because `position: sticky` only works within the boundaries of the parent, which is `div#application`. For the full source and text views, `change-view` regularly overflows outside of `div#application`. Without that rule, the navbar only stays fixed as long as the div is in view, but then scrolls away. I'm not sure why `change-view` overflows outside of the div, so if there's a better way to contain it I'd be happy to go with that method.

 In the meantime, `overflow: auto` works.

As a side-note, I guess those permanent scrollbars are only appearing in Safari because I don't see them in Chrome or Firefox. Unfortunately, I'm getting this error: `ReferenceError: Can't find variable: Intl` in my version of Safari (9.1.3), so I can't get the app to load at all which is why I haven't been seeing them. Looks like it's a API that doesn't exist in my old version. How should we go about polyfilling that?

Reply:
> I'm not sure why change-view overflows outside of the div, so if there's a better way to contain it I'd be happy to go with that method.

Oh, that’s because you are still using `height: 100%` on that view (so you can do flexbox, which, incidentally, would be a whole different way you could achieve this same thing). I should’ve looked and noticed that before.

> I guess those permanent scrollbars are only appearing in Safari because I don't see them in Chrome or Firefox

This is what it looks like in all browsers on OS X if you have a mouse attached or if you set your system settings to always show scrollbars (in the “general” panel). It’s the same in all browsers.

![scrolling](https://user-images.githubusercontent.com/74178/33257840-7deb36e2-d30c-11e7-86b3-e859d17acee0.png)

> I'm getting this error: ReferenceError: Can't find variable: Intl in my version of Safari (9.1.3), so I can't get the app to load at all which is why I haven't been seeing them. Looks like it's a API that doesn't exist in my old version. How should we go about polyfilling that?

So… there are probably a *lot* more things broken when you go back back more than two major releases. We haven’t really been doing the work to support anything that old. But either way, you could polyfill using https://www.npmjs.com/package/intl .  I would just skip it, though. Our date formatting isn’t that critical, I think, so you could just replace it with a call to `date.toLocaleString()` if `Intl` is not present.

Reply:
> you are still using height: 100% on that view

Ah, got it. Sorry, I'm still getting used to flexbox and how it interacts with other rules. I changed over to using flexbox to set the height of `#application`. To make it more sane, I rearranged `web-monitoring-ui`. To make sense of that crazy diff, basically all I did was get rid of the wrapping div around the router and placed the modal inside of `#application` instead. This allowed us to be rid of the extraneouse div and the `#web-monitoring-ui > div` rules. 

My only concern is if the modal has to exist outside of everything else for some reason, but it seems to be working fine. I also tested by commenting out this [check](https://github.com/lightandluck/web-monitoring-ui/blob/c292fb14963fcc53fa3faf59ba41333675d44e28/src/components/web-monitoring-ui.jsx#L126-L128). The modal still functions if a user is not logged in that case. Also, added a reminder that that check will need to be removed/updated when we get user permissions worked out.

Did not know that scrollbar setting existed. Glad I'm aware of it now! No wonder the scrolling rules were so much more aggravating for you then me. 😅

I'll open an issue for `Intl` and address it in another PR.

Reply:
>So… there are probably a lot more things broken when you go back back more than two major releases. We haven’t really been doing the work to support anything that old.

Yea, this is exactly one of the things I'd like to discuss in the suggested design meeting.

Reply:
> My only concern is if the modal has to exist outside of everything else for some reason, but it seems to be working fine.

It _did_. But the modal no longer renders where it looks like it renders (react-aria-modal yanks its element up outside the React app’s container), so the containing `div` is no longer necessary. :thumbsup: